syz-manager/manager.go

```diff
@@ -1367,7 +1376,16 @@ func (mgr *Manager) MachineChecked(features flatrpc.Feature, enabledSyscalls map
 				go mgr.dashboardReproTasks()
 			}
 		}
-		return queue.DefaultOpts(fuzzerObj, opts)
+		source := queue.DefaultOpts(fuzzerObj, opts)
+		if mgr.cfg.Snapshot {
+			log.Logf(0, "stopping VMs for snapshot mode")
+			mgr.source = source
+			mgr.loopStop()
```

Comment:
Alternatively, we could add some `Pool.SetDefault()` method that would stop all non-reserved VMs and restart them with a different default handler. At the start, it will be `mgr.fuzzerInstance` (like now), then you would replace it with some `mgr.snapshotFuzzerInstance`.

It would fit more organically into the current design and will provide a non-snapshotting-based bug reproduction out of box.

Reply:
Let's do it separately. Pool did not exist yet when I started this change.
I think we want reproduction to also use snapshot mode (more likely to reproduce the bug and is extremely faster). Also currently VMs used for snapshots can be stopped and reused for repro process.